### PR TITLE
kie-issues#384: Test Scenario Editor: Package and Dev webapp creation

### LINKED ISSUE #384: Test Scenario Editor: Package and Dev webapp creation
The scope of this ticket is to just create the Test Scenario package skeleton, with:
- A well-designed structure
- Min-required dependencies
- A Readme file
- The License
- A Dev webapp

### PR DESCRIPTION
**Pull request by yesamer**
Closes https://github.com/kiegroup/kie-issues/issues/384

This PR introduces the new generation of Test Scenario Editor (Scenario Simulation, `scesim`)

The aim is to lay the foundation of the Editor, just providing a well-structured project with a DevWebApp
A first "raw" version of the editor has been drafted.

Screenshot 2023-07-14 at 12 47 34

https://github.com/kiegroup/kie-tools/assets/16005046/a2df011c-33d1-4d4d-914b-bf68bb6f4216

### CODE REVIEW

**Review by tiagobento — COMMENTED** [review]
Nice! @yesamer Can you please just rename "Showcase" to "Dev webapp" I'm trying to remove this notion that these apps are just for show :D 

"Dev webapp" I think gives a better idea of what those apps are used for, and that they need to be maintained and taken care of.

**Comment by yesamer** [comment]
@tiagobento Renamed

File: packages/scesim-editor/dev-webapp/App.tsx [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+              Create a new Test Scenario
+            </Button>
+          </EmptyStateSecondaryActions>
+        </EmptyState>
+      )}
+      <div style={{ display: displayEditor() }}>
+        <div className="editor-footer ignore-onclickoutside">
+          <Split hasGutter={true}>
+            <SplitItem>
+              <Button variant="primary" onClick={() => setContent(undefined)} ouiaId="undo-button">
```
Comment (jomarko) [inline packages/scesim-editor/dev-webapp/App.tsx]:
```suggestion
              <Button variant="primary" onClick={() => setContent(undefined)} ouiaId="back-button">
```

File: packages/scesim-editor/src/editor/TestScenarioEditor.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+import * as React from "react";
+import { Notification } from "@kie-tools-core/notifications/dist/api";
+import { WorkspaceEdit } from "@kie-tools-core/workspace/dist/api";
+import { TextArea, TextInput } from "@patternfly/react-core/dist/js";
+
+interface Props {
+  /**
+   * Callback to the container so that it may bind to the TestScenarioEditor.
+   *
+   * @returns Instance of the TestScenarioEditor.
```
Comment (jomarko) [inline packages/scesim-editor/src/editor/TestScenarioEditor.tsx]: it seems `exposing` has `void` return type

**Review by jomarko — COMMENTED** [review]
Thank you for introducing this module @yesamer

**Review by tiagobento — APPROVED** [review]

**Review by myowinnaingg — APPROVED** [review]

**Pull request merged**